solidity/test/integration/WalleCreation.test.ts

```diff
@@ -0,0 +1,119 @@
+import type { BigNumberish, ContractTransaction } from "ethers"
+import type { SignerWithAddress } from "@nomiclabs/hardhat-ethers/signers"
+import type { Bridge, IRandomBeacon, WalletRegistry } from "../../typechain"
+
+import {
+  produceEcdsaDkgResult,
+  updateWalletRegistryDkgResultChallengePeriodLength,
+} from "./utils/ecdsa-wallet-registry"
+import { ecdsaWalletTestData } from "../data/ecdsa"
+import { produceRelayEntry } from "./utils/random-beacon"
+
+import { assertGasUsed } from "./utils/gas"
+import { fixture } from "./utils/fixture"
+
+const NO_MAIN_UTXO = {
+  txHash: "0x0000000000000000000000000000000000000000000000000000000000000000",
+  txOutputIndex: 0,
+  txOutputValue: 0,
+}
+
+const describeFn =
+  process.env.NODE_ENV === "integration-test" ? describe : describe.skip
+
+describeFn("Integration Test - Wallet Creation", async () => {
+  let bridge: Bridge
+  let walletRegistry: WalletRegistry
+  let randomBeacon: FakeContract<IRandomBeacon>
+  let governance: SignerWithAddress
+
+  const dkgResultChallengePeriodLength = 10
+
+  // TODO: Generate a random public key
+  const walletPublicKey = ecdsaWalletTestData.publicKey
+  const { walletID } = ecdsaWalletTestData
+  const walletPubKeyHash = ecdsaWalletTestData.pubKeyHash160
+
+  before(async () => {
+    ;({ governance, bridge, walletRegistry, randomBeacon } =
+      await waffle.loadFixture(fixture))
+
+    // Update only the parameters that are crucial for this test.
+    await updateWalletRegistryDkgResultChallengePeriodLength(
+      walletRegistry,
+      governance,
+      dkgResultChallengePeriodLength
+    )
+  })
+
+  describe("new wallet creation (happy path)", async () => {
+    let requestNewWalletTx: ContractTransaction
+    let walletRegistrationTx: ContractTransaction
+
+    before(async () => {
+      expect(await bridge.activeWalletPubKeyHash()).to.be.equal(
+        ethers.constants.AddressZero
+      )
+
+      requestNewWalletTx = await bridge.requestNewWallet(NO_MAIN_UTXO)
+      const startBlock = requestNewWalletTx.blockNumber
+
+      const relayEntry: BigNumberish = await produceRelayEntry(
+        walletRegistry,
+        randomBeacon
+      )
+
+      ;({ approveDkgResultTx: walletRegistrationTx } =
+        await produceEcdsaDkgResult(
+          walletRegistry,
+          walletPublicKey,
+          relayEntry,
+          startBlock
+        ))
+
+      await walletRegistrationTx.wait()
+    })
+
+    it("should register a new wallet in the WalletRegistry", async () => {
+      // eslint-disable-next-line @typescript-eslint/no-unused-expressions
+      expect(await walletRegistry.isWalletRegistered(walletID)).to.be.true
+    })
+
+    it("should register a new wallet details in the Bridge", async () => {
+      const storedWallet = await bridge.wallets(walletPubKeyHash)
+
+      expect(storedWallet.ecdsaWalletID).to.be.equal(
+        ecdsaWalletTestData.walletID
+      )
+
+      expect(storedWallet.state).to.be.equal(1)
```

Comment:
We can use:

https://github.com/keep-network/tbtc-v2/blob/4cd75f5a691a08de4eb56a16a27ec0f09e1008d4/solidity/test/fixtures/index.ts#L37-L44

Example:
https://github.com/keep-network/tbtc-v2/blob/4cd75f5a691a08de4eb56a16a27ec0f09e1008d4/solidity/test/bridge/Bridge.Wallets.test.ts#L608

Reply:
e438c884aca08cc30d869c7c7591460771bfa1c9